### PR TITLE
[SPARK-10271][Pyspark][MLLib] Added @since tags to pyspark.mllib.clustering

### PR DESCRIPTION
**Pull request by noel-smith**
Duplicated the @since decorator from pyspark.sql into pyspark (also tweaked to handle functions without docstrings).

Added @since to methods + "versionadded::" to classes (derived from the git file history in pyspark).

### CODE REVIEW

File: python/pyspark/mllib/clustering.py [hunk 1]
```diff
@@ -129,20 +135,32 @@ def computeCost(self, rdd):
...
+        """
+        Load a model from the given path.
+        """
         java_model = sc._jvm.org.apache.spark.mllib.clustering.KMeansModel.load(sc._jsc.sc(), path)
         return KMeansModel(_java2py(sc, java_model.clusterCenters()))
 
 
 class KMeans(object):
+    """
+    .. versionadded:: 0.9.1
```
Comment (noel-smith) [inline python/pyspark/mllib/clustering.py]: @davies Was `1.9.0-incubating` a public release? If so I'll update this to from `1.9.1` to `1.9.0`

Comment (davies) [inline python/pyspark/mllib/clustering.py]: All the releases are listed here: http://spark.apache.org/downloads.html

Comment (noel-smith) [inline python/pyspark/mllib/clustering.py]: Good point - I'll update to 1.9.2 to match public releases.

**Comment by noel-smith** [comment]
Jenkins, retest this please

**Comment by mengxr** [comment]
test this please

**Comment by mengxr** [comment]
@yu-iskw Could you help review this PR? Thanks!

**Comment by yu-iskw** [comment]
LGTM

**Comment by mengxr** [comment]
@noel-smith Could you merge current master and resolve conflicts?

**Comment by noel-smith** [comment]
Jenkins, retest this please

**Comment by noel-smith** [comment]
@yu-iskw Thanks for testing!
@mengxr I've rebased now - the changes were very minor. Should be good to merge once the tests complete.

**Comment by noel-smith** [comment]
Test this please

**Comment by mengxr** [comment]
add to whitelist

**Comment by mengxr** [comment]
test this please

**Comment by mengxr** [comment]
Merged into master. Thanks!

**Pull request closed without merging**